docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,302 @@
+
+Fetching data 
+=============
+
+The section below consists of a few lines of codes that help us download celebA dataset from a public web source and unzip them.
+
+.. code-block:: python
+
+    data_root = 'datasets'
+    base_url = "https://graal.ift.ulaval.ca/public/celeba/"
+
+    file_list = [
+        "img_align_celeba.zip",
+        "list_attr_celeba.txt", 
+        "identity_CelebA.txt",
+        "list_bbox_celeba.txt",
+        "list_landmarks_align_celeba.txt",
+        "list_eval_partition.txt"
+    ]
+
+    # Path to folder with the dataset
+    dataset_folder = f'{data_root}/celeba'
+    os.makedirs(dataset_folder, exist_ok=True)
+
+    for file in file_list:
+        url = f"{base_url}/{file}"
+        if not os.path.exists(f"{dataset_folder}/{file}"):
+        wget.download(url, f"{dataset_folder}/{file}")
+
+    with zipfile.ZipFile(f"{dataset_folder}/img_align_celeba.zip", 'r') as ziphandler:
+        ziphandler.extractall(dataset_folder)
+   
+Now, as the dataset id downloaded, we can define our datasets and dataloaders in its original way.
+
+.. code-block:: python
+
+    transforms = tfms.Compose ([
+                            tfms.Resize((image_size, image_size)),
+                            tfms.ToTensor(),
+                            tfms.Normalize(imagenet_mean, imagenet_std)
+    ])
+    train_dataset = datasets.CelebA(data_root, 
+                                    split='train', 
+                                    target_type=['attr', 'landmarks'], 
+                                    transform=transforms)
+    valid_dataset = datasets.CelebA(data_root, 
+                                    split='valid', 
+                                    target_type=['attr', 'landmarks'], 
+                                    transform=transforms)
+    test_dataset = datasets.CelebA(data_root, 
+                                split='test', 
+                                target_type=['attr', 'landmarks'], 
+                                transform=transforms)
+                                
+    train_dataloader = DataLoader(train_dataset, batch_size=batch_size, shuffle=True)
+    valid_dataloader = DataLoader(valid_dataset, batch_size=batch_size, shuffle=False)
+    test_dataloader = DataLoader(test_dataset, batch_size=batch_size, shuffle=False)
+
+Here we can see how each dataset sample looks like:
+
+.. code-block:: python
+
+    print (train_dataset[0])
+
+Regarding the complexity of the problem and the high number training/valid samples, we can seperate and use a portion of data as below:
+
+.. code-block:: python
+
+    train_subset = Subset(train_dataset, np.arange(1,10000))
+    valid_subset = Subset(valid_dataset, np.arange(1,2000))
+    train_dataloader = DataLoader(train_subset, batch_size=batch_size, shuffle=True)
+    valid_dataloader = DataLoader(valid_subset, batch_size=batch_size, shuffle=False)
```

Comment:
Not sure why we take a subset of the dataset. Is it because full training is too long?

Reply:
There is a huge number of images with a limited variation. (for example, eyes location don't vary very much). so using all images (189,643 images) is just time-consuming.

Reply:
Alright, maybe note that in the text.